persist/fs/equals_test.go

```diff
@@ -0,0 +1,93 @@
+package fs
+
+import (
+	indexfield "github.com/xichen2020/eventdb/index/field"
+)
+
+func stringFieldEquals(f1, f2 indexfield.StringField) (bool, error) {
+	// Asserts that two docs fields have equal values.
+	iter1, err := f1.Values().Iter()
+	if err != nil {
+		return false, err
+	}
+	iter2, err := f2.Values().Iter()
+	if err != nil {
+		return false, err
+	}
+	for iter1.Next() && iter2.Next() {
+		if iter1.Current() != iter2.Current() {
+			return false, nil
+		}
+	}
+	// If either iterator has extra data, they are not equal.
```

Comment:
Need to also check `iter1.Err()` and `iter2.Err()` are nil? Same for other functions in the file.

Reply:
You can just do `require.NoError(t, iter1.Err())` for example.